### DISCUSSION TITLE
Modify queries for both this page summary and all pages' summary

### DISCUSSION THREAD

**Discussion by saumanarsad** [filamentphp/filament#13709] [Ideas]
When we make a table using pagination and then add summary for columns, some columns may require us to use ->formatStateUsing. 
When we use this, the query returns an aggregate value and displays it as both This page and all pages' aggregate. Is there any way to modify the query for both separately?

**Comment by mc-fu** [comment]
I'm also searching for this. Did you ever find an answer to it?

**Comment by long-blade** [comment]
The only way i found is this: 

```php

Summarizer::make()
  ->label(__('storage.column.total_price'))
  ->numeric()
  ->money()
  ->using(function (Table $table, $livewire, $query) {
      // Shit fix, I don't know how to get the instance type
      $hasLimit = Str::contains($query->toSql(), 'limit');
      if ($hasLimit) {
          return $table->getRecords()->sum('total_price');
      }
  
      return $table->getQuery()->get()->sum('total_price');
})

```
